tests/test_objective_funs.py

```diff
@@ -485,25 +493,284 @@ def test_boundary_error_nestor(self):
+            copy=True,
+        )
+
+        np.testing.assert_allclose(things_with_eq_fixed[0].Phi_mn, 0, atol=1e-8)
+
+        # test 2 with field_fixed=True: the optimizer will zero out the nonaxisymmetric
+        # part of the equilibrium (Rb_lmn) since field is axisymmetric.
+        field = axisym_field.copy()
+        eq = nonaxisym_eq.copy()
+        field_grid, eval_grid, source_grid = get_grids(eq, field)
+
+        # fix modes where Rb_lmn = 0 so optimizer only pays attention to
+        # nonaxisymmetric modes
+        zero_indices = np.where(eq.Rb_lmn == 0)
+        fixed_R_modes = eq.surface.R_basis.modes[zero_indices]
+
+        zero_indices = np.where(eq.Zb_lmn == 0)
+        fixed_Z_modes = eq.surface.Z_basis.modes[zero_indices]
+
+        constraints = (
+            ForceBalance(eq),
+            FixPressure(eq),
+            FixCurrent(eq),
+            FixPsi(eq),
+            FixBoundaryR(eq, modes=fixed_R_modes),
+            FixBoundaryZ(eq, modes=fixed_Z_modes),
+        )
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+            field_fixed=True,
+        )
+        objective = ObjectiveFunction(quadflux_obj)
+
+        things_with_field_fixed, __ = nonlinear_optimizer.optimize(
+            eq,
+            objective=objective,
+            constraints=constraints,
+            gtol=0,
+            ftol=0,
+            copy=True,
+        )
+
+        new_Rb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+        new_Zb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+
+        np.testing.assert_allclose(new_Rb_lmn, 0, atol=1e-10)
+        np.testing.assert_allclose(new_Zb_lmn, 0, atol=1e-10)
+
+        # test with eq_fixed = field_fixed = False: nonaxisymmetric field and
+        # nonaxisymmetric equilibrium
+        eq = Equilibrium(M=2, L=2, N=4)
+        field = make_nonaxisym_field(eq)
+        field_grid, eval_grid, source_grid = get_grids(eq, field)
+        surface_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=20,
+            N=20,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+        )
+        plasma_vessel_obj = PlasmaVesselDistance(
+            surface=field,
+            eq=eq,
+            plasma_grid=source_grid,
+            surface_grid=surface_grid,
+            bounds=(0.2, 2),
+        )
+        mean_curvature_obj = MeanCurvature(eq, bounds=(-3, 3), grid=surface_grid)
+        objective = ObjectiveFunction(
+            (quadflux_obj, plasma_vessel_obj, mean_curvature_obj)
```

Comment:
`plasma_vessel_obj` is to prevent the equilibrium from intersecting the `field`? Why is `mean_curvature_obj` needed?

Reply:
mean curvature is needed because when you use the plasma vessel distance objective the errors are only evaluated point-wise, and no cost is given to just contorting the boundary shape to have the correct distances at the grid points but having a wild curvature or self-intersections in between.